exec/java-exec/src/main/java/org/apache/drill/exec/planner/sql/handlers/DropTableHandler.java

```diff
@@ -69,6 +71,15 @@ public PhysicalPlan getPlan(SqlNode sqlNode) throws ValidationException, RelConv
           .build(logger);
     }
 
+    final Table existingTable = SqlHandlerUtil.getTableFromSchema(drillSchema, tableName);
```

Comment:
I'm not sure about returning `null ` by `getTableFromSchema(...)` only if the table does not exist. Therefore renamed it to `tableToDrop ` and moved inside `if` block